src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageTest.cs

```diff
@@ -63,6 +64,31 @@ public void GetLanguageForInvariantCulture()
             ConfigTests.RunOnIsolatedThread(_defaultCulture, TestDefaultCulture);
         }
 
+        [Theory]
+        [InlineData("1234,5678", "#.##0,00", "1.234,57", "vi-VN")]
+        [InlineData("1234,5678", "#.##0,00", "1.234,57", "pt-BR")]
+        [InlineData("1234,5678", "# ##0,00", "1 234,57", "fr-FR")]
+        public void TextWithLanguageTest(string value, string format, string expectedResult,  string cultureName)
+        {
+            var culture = new CultureInfo(cultureName);
+            double.TryParse(value, NumberStyles.Float, culture, out double numValue);
+            NumberValue numberInput = new NumberValue(IRContext.NotInSource(FormulaType.Number), numValue);
+            StringValue formatString = new StringValue(IRContext.NotInSource(FormulaType.String), format); 
+
+            var formatInfo = new FormattingInfo()
+            {
+                CultureInfo = culture,
+                CancellationToken = CancellationToken.None,
+                TimeZoneInfo = TimeZoneInfo.Utc
+            };
+
+            FormulaValue[] args = { numberInput, formatString };
+
+            var result = Text(formatInfo, IRContext.NotInSource(FormulaType.String), args);
```

Comment:
>Text(formatInfo, IRContext.NotInSource(FormulaType.String), args) [](http://example.com/codeflow?start=25&length=65)

Instead of calling the function Text directly, consider going through the RecalcEngine, as the test below does.

Reply:
Updated to use recalcengine